### PR TITLE
fix: first ts file

### PR DESCRIPTION
**Pull request by doriaviram**

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -84,23 +89,26 @@ export const getTraceId = (awsXAmznTraceId) => {
     throw new Error(`Either Root, Parent or Sampled weren't found in traceId.`);
   }
 
+  // @ts-ignore
```
Comment (orr-levinger) [inline src/utils.ts:92]: traceId.Root!.split('-')[2]; i guess Root might be undefined

File: src/spans/awsSpan.ts [hunk 2]
```diff
@@ -326,6 +331,7 @@ export const getHttpSpan = (
   return { ...basicHttpSpan, info, service, started, ended };
 };
 
-export const addRttToFunctionSpan = (functionSpan, rtt) =>
+export const addRttToFunctionSpan = (functionSpan: any, rtt: number): any => {
```
Comment (orr-levinger) [inline src/spans/awsSpan.ts:334]: any?

File: src/utils.ts [hunk 3]
```diff
@@ -84,23 +89,26 @@ export const getTraceId = (awsXAmznTraceId) => {
...
   const transactionId = traceId.Root.split('-')[2];
 
+  // @ts-ignore
```
Comment (orr-levinger) [inline src/utils.ts:95]: missing type i guess

File: src/utils.ts [hunk 4]
```diff
@@ -84,23 +89,26 @@ export const getTraceId = (awsXAmznTraceId) => {
...
   traceId.transactionId = transactionId;
 
   return traceId;
 };
 
-export const getPatchedTraceId = (awsXAmznTraceId) => {
+export const getPatchedTraceId = (awsXAmznTraceId): string => {
+  // @ts-ignore
```
Comment (orr-levinger) [inline src/utils.ts:102]: why ignore?

**Review by GuyMoses — APPROVED** [review]

File: src/utils.ts [hunk 5]
```diff
@@ -472,16 +491,17 @@ export const md5Hash = (item) => {
...
     typeof encodingType === 'string' &&
     ['ascii', 'utf8', 'utf16le', 'ucs2', 'base64', 'binary', 'hex'].includes(encodingType)
   );
 
-export const isEmptyString = (str) => !!(!str || (typeof str === 'string' && str.length === 0));
+export const isEmptyString = (str): boolean =>
+  !!(!str || (typeof str === 'string' && str.length === 0));
 
-export const runOneTimeWrapper = (func, context) => {
+export const runOneTimeWrapper = (func: Function, context: any): Function => {
```
Comment (orr-levinger) [inline src/utils.ts:504]: any? maybe Record<string, string>

File: src/utils.ts [hunk 6]
```diff
@@ -492,4 +512,5 @@ export const runOneTimeWrapper = (func, context) => {
   };
 };
 
+// @ts-ignore
```
Comment (orr-levinger) [inline src/utils.ts:515]: why ignore?

File: src/utils.ts [hunk 7]
```diff
@@ -420,11 +439,11 @@ export const parseJsonFromEnvVar = (envVar, warnClient = false) => {
 };
 
 export const safeExecute = (
-  callback,
-  message = 'Error in Lumigo tracer',
-  logLevel = logger.LOG_LEVELS.WARNING,
-  defaultReturn = undefined
-) =>
+  callback: Function,
+  message: string = 'Error in Lumigo tracer',
```
Comment (orr-levinger) [inline src/utils.ts:443]: type is inferred automatically dont need to declare it

File: src/utils.ts [hunk 8]
```diff
@@ -406,11 +425,11 @@ const domainScrubbers = () =>
     (x) => new RegExp(x, 'i')
   );
 
-export const shouldScrubDomain = (url, domains = domainScrubbers()) => {
+export const shouldScrubDomain = (url, domains = domainScrubbers()): boolean => {
   return !!url && domains.some((regex) => url.match(regex));
 };
 
-export const parseJsonFromEnvVar = (envVar, warnClient = false) => {
+export const parseJsonFromEnvVar = (envVar, warnClient = false): {} | undefined => {
```
Comment (orr-levinger) [inline src/utils.ts:432]: {}? maybe use object or any?

File: src/utils.ts [hunk 9]
```diff
@@ -395,7 +413,8 @@ export const getJSONBase64Size = (obj) => {
 export const parseQueryParams = (queryParams) => {
   if (typeof queryParams !== 'string') return {};
   let obj = {};
-  queryParams.replace(/([^=&]+)=([^&]*)/g, function (m, key, value) {
+  // @ts-ignore
```
Comment (orr-levinger) [inline src/utils.ts:416]: why ignore?

File: src/utils.ts [hunk 10]
```diff
@@ -198,46 +207,55 @@ export const isVerboseMode = () => validateEnvVar(VERBOSE_FLAG);
...
 
-export const getHandlerContext = () => TracerGlobals.getHandlerInputs().context || {};
-
 export const getMaxRequestSize = () => TracerGlobals.getTracerInputs().maxSizeForRequest;
 
-export const getInvokedArn = () => getHandlerContext().invokedFunctionArn || '';
-export const getInvokedVersion = () => getHandlerContext().functionVersion || '';
+export const getInvokedArn = () => {
+  // @ts-ignore
+  return TracerGlobals.getHandlerInputs().context.invokedFunctionArn || '';
```
Comment (orr-levinger) [inline src/utils.ts:252]: probably need ! or ?
TracerGlobals.getHandlerInputs().context?.invokedFunctionArn?

**Review by OrrLumigo — APPROVED** [review]

**Pull request merged**